frontier/code/modules/projectiles/guns/energy.dm

```diff
@@ -0,0 +1,84 @@
+/obj/item/weapon/gun/energy
+    var/bat_insert_sound = 'sound/weapons/guns/interaction/pistol_magin.ogg'
+    var/bat_remove_sound = 'sound/weapons/guns/interaction/pistol_magout.ogg'
+    var/hatch = TRUE
+
+/obj/item/weapon/gun/energy/examine(mob/user)
+    . = ..(user)
+    if (!cell_type)
+        to_chat(user, "This weapon uses custom energy cell as a power source.")
+    else
+        switch(cell_type)
+            if (/obj/item/weapon/cell/device/weapon/normal)
+                to_chat(user, "This weapon uses a normal-size energy cell as a power source.")
+            if (/obj/item/weapon/cell/device/weapon/small)
+                to_chat(user, "This weapon uses a small energy cell as a power source.")
+            if (/obj/item/weapon/cell/device/weapon/big)
+                to_chat(user, "This weapon uses a big energy cell as a power source.")
+
+/obj/item/weapon/gun/energy/proc/unload_battary(mob/user)
+    if(cell_type == null || use_external_power)
+        to_chat(user, "<span class='warning'>You can't pull the battery out of [src].</span>")
+        return
+    if(!hatch)
+        to_chat(user, "<span class='warning'>Open the hatch first.</span>")
+        return
+    if (power_supply)
+        if(do_after(user, 7 * (SKILL_MAX + 1 - user.get_skill_value(SKILL_WEAPONS)) , src))
+            user.put_in_hands(power_supply)
+            user.visible_message("[user] removes [power_supply] from [src].", "<span class='notice'>You remove [power_supply] from [src].</span>")
```

Comment:
SPAN_NOTICE()

Reply:
Not absolutely necessary but you can open a PR changing these if you like to